src/components/remote_control/include/remote_control/remote_control_plugin.h

```diff
@@ -97,25 +97,48 @@ class RemoteControlPlugin : public RemotePluginInterface {
    */
   void OnUnregisterApplication(const uint32_t app_id) OVERRIDE;
 
+  /**
+   * @brief Sends HMI status notification to mobile
+   * @param app application with changed HMI status
+   **/
   void SendHmiStatusNotification(
       application_manager::ApplicationSharedPtr app) OVERRIDE;
 
+  /**
+   * @brief Getter for event_dispatcher
+   * @return reference to RCEventDispatcher instance
+   */
   RCEventDispatcher& event_dispatcher() OVERRIDE;
 
+  /**
+   * @brief Getter for resource_allocation_manager
+   * @return reference to ResourceAllocationManager instance
+   */
   ResourceAllocationManager& resource_allocation_manager() OVERRIDE;
 
+  /**
+   * @brief Overriden setter for service
+   * @param service pointer to new service instance
+   */
+  void set_service(application_manager::ServicePtr service) OVERRIDE;
```

Comment:
@AKalinich-Luxoft why you need this interface? I Don't see calling it except unit tests

Reply:
@LuxoftAKutsan it is called from `plugin_manager.cc` line 119 when plugins is loading. Exactly this implementation will be called on `RemoteControlPlugin` instance loading.
In `RemoteControlPlugin` we need to access AM settings to get timeout value for RC request controller, however it could be done only through `Service` object which is set on `set_serice()` call from `plugin_manager`.

Reply:
@AKalinich-Luxoft ok